cmd/rm-main.go

```diff
@@ -428,37 +428,55 @@ func listAndRemove(url string, opts removeOpts) error {
 		if opts.nonCurrentVersion && opts.isRecursive && opts.withVersions {
 			if lastPath != content.URL.Path {
 				lastPath = content.URL.Path
-				if isNonCurrent(perObjectVersions) {
+				for _, content := range perObjectVersions {
+					if content.IsLatest && !isLatestDeleteMarker(perObjectVersions) {
```

Comment:
the isLatestDeleteMarker check ensures we remove the delete marker if it is topmost in the stack of versions. However, if top most is an object version we want to avoid removing it

Reply:
Isnt the first occurrence of the object the latest one? why do we need to collect in perObjectVersions?

Reply:
we are simply skipping the latest version, except when it is a delete marker

Reply:
okay, then maybe we can revert the change in the --non-current flag description

Reply:
@vadmeste, the --non-current flag was implemented originally to allow version unaware applications to clean up the version stack when they did a `mc rm` to set a delete marker. The --non-current flag previously applied only when delete marker was the top most version.

 This PR changes this to remove --non-current versions irrespective of whether top most is a delete marker or not. When top most is a delete marker, it does not make sense to leave it behind as the underlying object versions are gone - hence it will be deleted as well.